internal/controllers/clusterrequest_controller.go

```diff
@@ -652,26 +695,34 @@ func (t *clusterRequestReconcilerTask) getCrClusterTemplateRef(ctx context.Conte
 	return clusterTemplateRef, nil
 }
 
-// validateClusterTemplateInput validates if the clusterTemplateInput matches the
+// validateClusterTemplateInputMatchesSchema validates if the clusterTemplateInput matches the
 // inputDataSchema of the ClusterTemplate
-func (t *clusterRequestReconcilerTask) validateClusterTemplateInputMatchesClusterTemplate(
-	ctx context.Context, clusterTemplateRef *oranv1alpha1.ClusterTemplate, clusterInstanceData map[string]any) error {
+func (t *clusterRequestReconcilerTask) validateClusterTemplateInputMatchesSchema(
+	ctx context.Context,
+	clusterTemplateRef *oranv1alpha1.ClusterTemplate,
+	clusterData map[string]any,
+	schemaType string) error {
 
 	// Get the input data in string format.
-	jsonString, err := json.Marshal(clusterInstanceData)
+	jsonString, err := json.Marshal(clusterData)
 	if err != nil {
 		t.logger.ErrorContext(
 			ctx,
 			fmt.Sprintf(
-				"Error Marshaling the provided clusterInstance data from "+
+				"Error Marshaling the provided cluster data from "+
 					"ClusterTemplate %s for ClusterRequest %s",
 				t.object.Spec.ClusterTemplateRef, jsonString))
 		return err
 	}
 
 	// Check that the clusterTemplateInput matches the inputDataSchema from the ClusterTemplate.
-	err = utils.ValidateJsonAgainstJsonSchema(
-		string(clusterTemplateRef.Spec.InputDataSchema.ClusterInstanceSchema.Raw), string(jsonString))
+	if schemaType == utils.ClusterInstanceSchema {
+		err = utils.ValidateJsonAgainstJsonSchema(
+			string(clusterTemplateRef.Spec.InputDataSchema.ClusterInstanceSchema.Raw), string(jsonString))
```

Comment:
I thought the clusterInstanceSchema was now using YAML? Should ValidateJsonAgainstJsonSchema be renamed?